File: runtime/cpp/cyana-ast/CyanaAstRuntimeException.cpp
//
// Created by tian wei jun on 2022/11/23 0023.
//

#include "CyanaAstRuntimeException.h"

CyanaAstRuntimeException::CyanaAstRuntimeException(std::string msg) noexcept : msg(msg) {
}

CyanaAstRuntimeException::~CyanaAstRuntimeException() noexcept {
}

const char *CyanaAstRuntimeException::what() const noexcept {
  return msg.c_str();
}

File: runtime/java/cyana-ast-runtime/src/cyana/ast/runtime/CyanaAstRuntimeException.java
package cyana.ast.runtime;

/**
 * 自定义运行时异常.
 *
 * @author tian wei jun
 */
public class CyanaAstRuntimeException extends RuntimeException {

  public CyanaAstRuntimeException(Throwable cause) {
    super(cause);
  }

  public CyanaAstRuntimeException(String message) {
    super(message);
  }
}
